### PR TITLE
Add cattle cluster agent cpu and mem request recommendation

### PR DESCRIPTION
**Pull request by btat**
SURE-7353

## Description

Add a recommendation for baseline CPU and memory request values for the cattle agent using values specified in parent ticket.

### CODE REVIEW

File: versioned_docs/version-2.7/reference-guides/best-practices/rancher-managed-clusters/tips-to-set-up-containers.md [hunk 1]
```diff
@@ -48,9 +52,10 @@ If you set a resource limit for the namespace that the pod is deployed in, and t
 
 :::
 
-It is recommended to define resource requirements on the container level because otherwise, the scheduler makes assumptions that will likely not be helpful to your application when the cluster experiences load.
+As a starting point, we recommend setting your CPU request at 50m and memory request at 100Mi. You should choose your limits based on the resources available in your environment and your use case with Rancher. It is recommended to define resource requirements on the container level because otherwise, the scheduler makes assumptions that will likely not be helpful to your application when the cluster experiences load.
```
Comment (moio) [inline versioned_docs/version-2.7/reference-guides/best-practices/rancher-managed-clusters/tips-to-set-up-containers.md:55]: This page seems to be about general containers, but the intention is to explain about the specific Pod Rancher automatically installs to Rancher registered clusters (cattle-cluster-agent).

See also https://jira.suse.com/browse/SURE-7353?focusedId=1344617&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-1344617

**Review by moio — CHANGES_REQUESTED** [review]

File: versioned_docs/version-2.7/reference-guides/best-practices/rancher-managed-clusters/tips-to-set-up-containers.md (hunk 1 above)
Comment (btat) [reply versioned_docs/version-2.7/reference-guides/best-practices/rancher-managed-clusters/tips-to-set-up-containers.md:55]: Ah ok, I'll make some adjustments. Thanks Silvio.

File: docs/reference-guides/best-practices/rancher-managed-clusters/tips-to-set-up-containers.md [hunk 2]
```diff
@@ -50,7 +54,10 @@ If you set a resource limit for the namespace that the pod is deployed in, and t
 
 It is recommended to define resource requirements on the container level because otherwise, the scheduler makes assumptions that will likely not be helpful to your application when the cluster experiences load.
 
+TODO we recommend to set your CPU request at 50m and memory request at 100Mi, however this is a base recommendation and customers should choose based on resources available in their environment and their use case on how they use Rancher.
```
Comment (asettle) [inline docs/reference-guides/best-practices/rancher-managed-clusters/tips-to-set-up-containers.md:57]: I don't think I phrased this particularly well, how about:

_As a base line we recommend setting the CPU request at 50m and memory request at 100Mi. However, it is important that you ensure your cluster has the correct resources available and the use case has been assessed appropriately before doing so._

File: versioned_docs/version-2.7/reference-guides/best-practices/rancher-managed-clusters/tips-to-set-up-containers.md (hunk 1 above)
Comment (asettle) [reply versioned_docs/version-2.7/reference-guides/best-practices/rancher-managed-clusters/tips-to-set-up-containers.md:55]: I've reworded it, as suggested above: 

_As a base line we recommend setting the CPU request at 50m and memory request at 100Mi. However, it is important that you ensure your cluster has the correct resources available and the use case has been assessed appropriately before doing so._

Comment (moio) [reply versioned_docs/version-2.7/reference-guides/best-practices/rancher-managed-clusters/tips-to-set-up-containers.md:55]: @asettle my remark was more about the position of this remark in the documentation. SURE-7353 is not about any general container, it is specific to the cattle-cluster-agent pod Rancher installs in downstream clusters to manage them.

Is the intention to move this is some place more specific? This page seems to be about general recommendations for any containerized workload?

Comment (btat) [reply versioned_docs/version-2.7/reference-guides/best-practices/rancher-managed-clusters/tips-to-set-up-containers.md]: @moio I moved the info to the [dedicated Rancher agent page](https://ranchermanager.docs.rancher.com/how-to-guides/new-user-guides/launch-kubernetes-with-rancher/about-rancher-agents) that we have. Let me know if that's better.

**Review by moio — APPROVED** [review]
lgtm

File: docs/how-to-guides/new-user-guides/launch-kubernetes-with-rancher/about-rancher-agents.md [hunk 3]
```diff
@@ -21,11 +21,62 @@ The `cattle-cluster-agent` is used to connect to the Kubernetes API of [Rancher
 
 The `cattle-node-agent` is used to interact with nodes in a [Rancher Launched Kubernetes](launch-kubernetes-with-rancher.md) cluster when performing cluster operations. Examples of cluster operations are upgrading Kubernetes version and creating/restoring etcd snapshots. The `cattle-node-agent` is deployed using a DaemonSet resource to make sure it runs on every node. The `cattle-node-agent` is used as fallback option to connect to the Kubernetes API of [Rancher Launched Kubernetes](launch-kubernetes-with-rancher.md) clusters when `cattle-cluster-agent` is unavailable.
 
+### Requests
+
+The `cattle-cluster-agent` pod does not define default CPU and memory request values. As a baseline, we recommend setting the CPU request at `50m` and memory request at `100Mi`. However, it is important that you ensure your cluster has the correct resources available and the use case has been assessed appropriately before doing so.
```
Comment (martyav) [inline docs/how-to-guides/new-user-guides/launch-kubernetes-with-rancher/about-rancher-agents.md:26]:
```suggestion
The `cattle-cluster-agent` pod does not define the default CPU and memory request values. As a baseline, we recommend setting the CPU request at `50m` and memory request at `100Mi`. However, it is important that you assess your use case appropriately and that you allocate the correct resources to your cluster for your distinct needs.
```

File: docs/how-to-guides/new-user-guides/launch-kubernetes-with-rancher/about-rancher-agents.md [hunk 4]
```diff
@@ -21,11 +21,62 @@ The `cattle-cluster-agent` is used to connect to the Kubernetes API of [Rancher
...
+
+To configure request values through the UI:
+
+<Tabs groupId="k8s-distro">
+<TabItem value="RKE">
+
+1. When you create or edit an existing cluster, go to the **Cluster Options** section.
```
Comment (martyav) [inline docs/how-to-guides/new-user-guides/launch-kubernetes-with-rancher/about-rancher-agents.md:33]: Maybe link to how to create/edit a cluster?

File: docs/how-to-guides/new-user-guides/launch-kubernetes-with-rancher/about-rancher-agents.md [hunk 5]
```diff
@@ -21,11 +21,62 @@ The `cattle-cluster-agent` is used to connect to the Kubernetes API of [Rancher
...
+</TabItem>
+</Tabs>
+
 ### Scheduling rules
 
-The `cattle-cluster-agent` uses either a fixed set of tolerations, or dynamically-added tolerations based on taints applied to the control plane nodes. This structure allows [Taint based Evictions](https://kubernetes.io/docs/concepts/scheduling-eviction/taint-and-toleration/#taint-based-evictions) to work properly for `cattle-cluster-agent`. 
+The `cattle-cluster-agent` uses either a fixed set of tolerations, or dynamically-added tolerations based on taints applied to the control plane nodes. This structure allows [Taint based Evictions](https://kubernetes.io/docs/concepts/scheduling-eviction/taint-and-toleration/#taint-based-evictions) to work properly for `cattle-cluster-agent`.
 
-If control plane nodes are present in the cluster, the default tolerations will be replaced with tolerations matching the taints on the control plane nodes. The default set of tolerations are described below. 
+If control plane nodes are present in the cluster, the default tolerations will be replaced with tolerations matching the taints on the control plane nodes. The default set of tolerations are described below.
```
Comment (martyav) [inline docs/how-to-guides/new-user-guides/launch-kubernetes-with-rancher/about-rancher-agents.md:79]:
```suggestion
If control plane nodes are present in the cluster, the default tolerations are replaced with tolerations matching the taints on the control plane nodes. The default set of tolerations are described below.
```

**Review by martyav — CHANGES_REQUESTED** [review]
I made some suggestions for the file in /docs

File: docs/how-to-guides/new-user-guides/launch-kubernetes-with-rancher/about-rancher-agents.md (hunk 3 above)
Comment (btat) [reply docs/how-to-guides/new-user-guides/launch-kubernetes-with-rancher/about-rancher-agents.md]: Updated in 8b71096b32e125601735361546d2781c552d0d28.

File: docs/how-to-guides/new-user-guides/launch-kubernetes-with-rancher/about-rancher-agents.md (hunk 4 above)
Comment (btat) [reply docs/how-to-guides/new-user-guides/launch-kubernetes-with-rancher/about-rancher-agents.md]: Updated in 8b71096b32e125601735361546d2781c552d0d28. Added a link for creation, but we don't have a generic 'edit cluster' page to reference.

**Review by martyav — APPROVED** [review]

**Pull request merged**